crates/viewer/re_blueprint_tree/src/blueprint_tree.rs

```diff
@@ -576,11 +642,111 @@ impl BlueprintTree {
             ItemContext::BlueprintTree {
                 filter_session_id: self.filter_state.session_id(),
             },
-            &response,
+            response,
             SelectionUpdateBehavior::UseSelection,
         );
-        self.scroll_to_me_if_needed(ui, &item, &response);
-        ctx.handle_select_hover_drag_interactions(&response, item, true);
+        self.scroll_to_me_if_needed(ui, &item, response);
+        ctx.handle_select_hover_drag_interactions(response, item.clone(), true);
+
+        self.handle_range_selection(ctx, blueprint_tree_data, item, response);
+    }
+
+    /// Handle setting/extending the selection based on shift-clicking.
+    fn handle_range_selection(
+        &mut self,
+        ctx: &ViewerContext<'_>,
+        blueprint_tree_data: &BlueprintTreeData,
+        item: Item,
+        response: &Response,
+    ) {
+        // Early out if we're not being clicked.
+        if !response.clicked() {
+            return;
+        }
+
+        let modifiers = ctx.egui_ctx.input(|i| i.modifiers);
+
+        if modifiers.shift {
+            if self.last_clicked_item.is_some() {
+                let items_iterator = self
+                    .items_in_range(ctx, blueprint_tree_data, &item)
+                    .into_iter()
+                    .map(|item| {
+                        (
+                            item,
+                            Some(ItemContext::BlueprintTree {
+                                filter_session_id: self.filter_state.session_id(),
+                            }),
+                        )
+                    });
+
+                if items_iterator.len() > 0 {
+                    if modifiers.command {
+                        ctx.selection_state.extend_selection(items_iterator);
+                    } else {
+                        ctx.selection_state.set_selection(items_iterator);
+                    }
+                }
+            }
+        } else {
+            self.last_clicked_item = Some(item);
```

Comment:
I feel like `last_clicked_item` is a bit of a misnomer since after a shift-click it is not the last clicked but more like `start_of_range_selection`, but I figure that's also confusing since when we first set it we don't know yet if that is going to be a range selection. Hrrrrm :(

Maybe `range_selection_anchor` or something?

Reply:
`range_selection_anchor_item` it is :+1:

Reply:
`range_selection_anchor_item` it is :+1: